stubs/Flask-SocketIO/flask_socketio/__init__.pyi

```diff
@@ -0,0 +1,85 @@
+from collections.abc import MutableMapping
+from threading import Thread
+from typing import Any, Callable, Literal, TypeVar
+from typing_extensions import TypeAlias
+
+import pywsgi
+import socketio
+from flask import Flask
+from flask.sessions import SessionMixin
+from flask.testing import FlaskClient
+from socketio import socketio_manage as socketio_manage
+
+from .namespace import Namespace
+from .test_client import SocketIOTestClient
+
+_ExceptionHandler = TypeVar("_ExceptionHandler", Callable)
+_Handler = TypeVar("Handler", Callable)
+_Environ = TypeVar("Environ", MutableMapping[str, Any])
+
+_HandlerDecorator: TypeAlias = Callable[_Handler, _Handler]
+
+gevent_socketio_found: bool
+
+class _SocketIOMiddleware(socketio.WSGIApp):
+    flask_app: Flask
+    def __init__(self, socketio_app: socketio.Server, flask_app: Flask, socketio_path: str = "socket.io") -> None: ...
+    def __call__(self, environ: _Environ, start_response: Callable[[str, list[tuple[str, str]]]]): ...
+
+class _ManagedSession(dict, SessionMixin): ...
+
+class SocketIO:
+    server: socketio.Server | None
+    server_options: dict[str, Any]
+    wsgi_server: pywsgi.WSGIServer | None
+    handlers: list[_Handler]
+    namespace_handlers: list
+    exception_handlers: list[_ExceptionHandler]
+    default_exception_handler: None | _ExceptionHandler
+    manage_session: bool
```

Comment:
None of the attributes in lines 32-39 are public nor documented. My opinion is that they should not be included here. I consider these internal, so they can change at any time without notice.

Reply:
If these attributes are removed, a comment should be added stating why they are intentionally omitted from the stub. Otherwise it's possible somebody will come along in the future complaining about them being missing from the stub.

Reply:
@miguelgrinberg Removed these private attributes